internal/factsengine/factsengine.go

```diff
@@ -204,36 +229,26 @@ func gatherFacts(
 	return factsResults, nil
 }
 
-func parseFactsRequest(request []byte) (*gatherers.GroupedFactsRequest, error) {
-	var factsRequest gatherers.FactsRequest
+func parseFactsRequest(event *contracts.FactsGatheringRequestedV1) (*gatherers.GroupedFactsRequest, error) {
 	var groupedFactsRequest *gatherers.GroupedFactsRequest
 
-	// the following event expects is suitable for the one message only scenario
-	// event, err := contracts.NewFactsGatheringRequestedV1FromJsonCloudEvent(request)
-
-	// hack to get only the needed data, this should be encapsulated in contracts repo
-	var result map[string]interface{}
-
-	err := json.Unmarshal(request, &result)
-	if err != nil {
-		return nil, err
-	}
-
-	request, _ = json.Marshal(result["data"])
-
-	err = json.Unmarshal(request, &factsRequest)
-	if err != nil {
-		return nil, err
-	}
-
 	groupedFactsRequest = &gatherers.GroupedFactsRequest{
-		ExecutionID: factsRequest.ExecutionID,
+		ExecutionID: event.ExecutionId,
 		Facts:       make(map[string][]gatherers.FactRequest),
 	}
 
 	// Group the received facts by gatherer type, so they are executed in the same moment with the same source of truth
-	for _, factRequest := range factsRequest.Facts {
-		groupedFactsRequest.Facts[factRequest.Gatherer] = append(groupedFactsRequest.Facts[factRequest.Gatherer], factRequest)
+	for _, agentFacts := range event.Facts {
```

Comment:
I am realizing two things here:
- `FactsGatheringRequestedV1.Facts` could also be renamed to `FactsGatheringRequestedV1.AgentsFacts` (or similar), as it is actually a collection of objects representing the `Facts` to be gathered for an `AgentId`.

- `FactsGatheringRequestedV1.Targets` might not be really helping much as it requires us to iterate twice.
Removing it would allow us to iterate only once over the `AgenstFacts`, checking if interested and getting the needed information.